packages/payment-processor/src/payment/erc20.ts

```diff
@@ -55,18 +66,61 @@ export async function hasErc20Approval(
   account: string,
   provider: Provider = getNetworkProvider(request),
 ): Promise<boolean> {
-  const erc20Contract = ERC20Contract.connect(request.currencyInfo.value, provider);
-  const allowance = await erc20Contract.allowance(account, getProxyAddress(request));
-  return allowance.gt(request.expectedAmount);
+  return checkErc20Allowance(
+    account,
+    getProxyAddress(request),
+    provider,
+    request.currencyInfo,
+    new BigNumber(request.expectedAmount)
+  )
+}
+
+/**
+ * Checks if a spender has enough allowance from an ERC20 token owner to pay an amount.
+ * @param ownerAddress address of the owner
+ * @param spenderAddress address of the spender
+ * @param provider the web3 provider. Defaults to Etherscan.
+ * @param paymentCurrency ERC20 currency
+ * @param amount
+ */
+export async function checkErc20Allowance(
```

Comment:
Nice change. But mixing allowance and approval can be confusing. I think we should stick to only one name.

Reply:
Naming-wise, I tried my best to circumvent all the previous asumptions that we pay in request currency. For example `hasErc20Approval` should be renamed `hasRequestErc20Approval` or `hasDefaultErc20Approval` or something like that.

Do you prefer `hasAnyErc20Approval` ?